src/data/mystery-encounters/encounters/dancing-lessons-encounter.ts

```diff
@@ -271,9 +271,9 @@ export const DancingLessonsEncounter: MysteryEncounter = MysteryEncounterBuilder
         const move = encounter.misc.selectedMove?.getMove().id;
         if (!oricorio.moveset.some((m) => m.getMove().id === move)) {
           if (oricorio.moveset.length < 4) {
-            oricorio.moveset.push(new PokemonMove(move));
+            oricorio.moveset.push(new PokemonMove(move, { pokemonId: oricorio }));
```

Comment:
Ah... somehow everything is typed as `any` here... thanks MEs.
```suggestion
            oricorio.moveset.push(new PokemonMove(move, { pokemonId: oricorio.id }));
```

Reply:
Ah, so this is still accessing the moveset directly as well...
Fixing that: [001-dancing-lesson-me-moveset.diff.txt](https://github.com/user-attachments/files/21663823/001-dancing-lesson-me-moveset.diff.txt)